### ISSUE TITLE
HelloUWP and similarly written apps won't show the first view when the app relaunched while already running

### ISSUE THREAD

**Issue by balazsgerlei** [Caliburn-Micro/Caliburn.Micro#327]
Hello!
I'm trying to write an UWP app using Caliburn.Micro 3.0.1 for Windows 10 PC and Mobile with quite similar base structure to the Hello UWP sample. I noticed that when my app is running in the background and I relaunch it from the app list or it's tile it wont display the first ViewModel inside ShellViewModel.

I tried out with the Hello UWP sample too, but it does the same thing. Won't launch when already running in the background. Debugging it shows that the ShellViewModel launches but the first ViewModel displayed inside it won't.

This is the sample I'm refering to: [link](https://github.com/Caliburn-Micro/Caliburn.Micro/tree/master/samples/Caliburn.Micro.HelloUWP)

Please help me, this is quite a big and urgent problem considering my app.

**Edit:**
I tried out, and the same occurs on Windows Desktop... :(

**Comment by nigel-sampson** [comment]
I'm assuming you've launched the app and already navigated to that other view, and then on relaunch you're not seeing it?

**Comment by balazsgerlei** [comment]
Yeah, I do the exact thing you described and then (sticking with the HelloUWP project, but both that and my app behaves the same in this regard) it shows the view for the ShellViewModel and not for the DeviceViewModel. Even trying to load the latter view with the menu item in the "hamburger" menu have no effect.

I checked and when doing this, in the App.xaml.cs's OnLaunch method the LaunchActivatedEventArgs args parameters' PreviousExecutionState equals to ApplicationExecutionState.Running (which is not suprising since it is running).

**Comment by nigel-sampson** [comment]
Ok, so for the first problem, I'd check of if the `PreviousExecutionState` is `Running` and skip the call to `DisplayRootViewFor<ShellViewModel>();` if it is.

I'd also check the sample to see if it has [these lines](https://github.com/Caliburn-Micro/Caliburn.Micro/blob/master/samples/Caliburn.Micro.HelloUWP/Caliburn.Micro.HelloUWP/ViewModels/ShellViewModel.cs#L32) which were added recently.

Can't add much more tonight without a machine to test on but that should help.

**Comment by balazsgerlei** [comment]
Yeah, that was it!
The two lines you refered to solved the problem! Sorry I haven't noticed I ran an older version of the sample.

**Comment by balazsgerlei** [comment]
One more question though. Should I also skip the call to DisplayRootViewFor when PreviousExecutionState is Running? Or it was just a suggestion if the other one does not work?

**Comment by balazsgerlei** [comment]
Ok, I tested it and now I know why you said that! If I call DisplayRootViewFor<ShellViewModel> it will display the home screen of my app instead of the screen that the user was on when left the app. It's definitely better to only call DisplayRootViewFor when the app was not running!

Thank you for your quick response, it was a critical bug in my app and I'm really glad that it was solved so quickly! :)
Keep up the good work on this amasing MVVM framework!
